### PR TITLE
Better error handling and logging for get operation status

### PR DESCRIPTION
**Pull request by honourfish**
## Overview
* Catch and handle http exceptions by logging them and carrying on the polling
* Introduce logging and log levels

## Testing

Failure with log level `WARNING`:
```
python3 ./scitt/check_operation_status.py --operation-id foo
signed statement not registered within polling duration
```

Failure with log level `DEBUG`:
```
python3 ./scitt/check_operation_status.py --operation-id foo --log-level DEBUG
INFO:check operation status:starting to poll for operation status 'succeeded'
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): app.datatrails.ai:443
DEBUG:urllib3.connectionpool:https://app.datatrails.ai:443 "GET /archivist/v1/publicscitt/operations/foo HTTP/1.1" 400 47
DEBUG:check operation status:failed getting operation status, error: 400 Client Error: Bad Request for url: https://app.datatrails.ai/archivist/v1/publicscitt/operations/foo
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): app.datatrails.ai:443
DEBUG:urllib3.connectionpool:https://app.datatrails.ai:443 "GET /archivist/v1/publicscitt/operations/foo HTTP/1.1" 400 47
DEBUG:check operation status:failed getting operation status, error: 400 Client Error: Bad Request for url: https://app.datatrails.ai/archivist/v1/publicscitt/operations/foo
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): app.datatrails.ai:443
DEBUG:urllib3.connectionpool:https://app.datatrails.ai:443 "GET /archivist/v1/publicscitt/operations/foo HTTP/1.1" 400 47
DEBUG:check operation status:failed getting operation status, error: 400 Client Error: Bad Request for url: https://app.datatrails.ai/archivist/v1/publicscitt/operations/foo
...
```

### CODE REVIEW

File: scitt/check_operation_status.py [hunk 1]
```diff
@@ -91,11 +104,22 @@ def main():
...
 
+    # log level
+    parser.add_argument(
+        "--log-level",
+        type=str,
+        help="log level. for any individual poll errors use DEBUG, defaults to WARNING",
+        default="WARNING",
+    )
+
     args = parser.parse_args()
```
Comment (robinbryce) [inline scitt/check_operation_status.py:116]: Why not have a run() method that main calls then do the usual idiomatic sinlge line exception printing for example,

```
try:
    # Simulate an error
    1 / 0
except Exception as e:
    # Format the traceback as a single line
    tb_str = ''.join(traceback.format_exception(etype=type(e), value=e, tb=e.__traceback__)).replace('\n', ' ')
    # Print to stderr
    print(tb_str, file=sys.stderr)
```

**Review by robinbryce — CHANGES_REQUESTED** [review]
I think we would do much better to hide the python specific traceback sytax

File: scitt/check_operation_status.py (hunk 1 above)
Comment (robinbryce) [reply scitt/check_operation_status.py:116]: It is not unusual to exose the raw traceback when the user has explicitly set loglevel DEBUG. the loggin package allows this to be introspected I beleive

File: scitt/check_operation_status.py [hunk 2]
```diff
@@ -41,20 +42,32 @@ def get_operation_status(operation_id: str, headers: dict) -> dict:
...
+            # pylint: disable=fixme
+            # TODO: ensure get_operation_status handles error cases from the rest request
+            if (
+                "status" in operation_status
+                and operation_status["status"] == "succeeded"
+            ):
+                return operation_status["entryID"]
+
+        except requests.HTTPError as e:
+            logger.debug(f"failed getting operation status, error: {e}")
```
Comment (eccles) [inline scitt/check_operation_status.py:70]: its inefficient to use f strings - this ode will format the string regardles of whether the logging level is set. Use old-style C format such as %s instead.

https://docs.python.org/3/howto/logging.html#logging-variable-data

File: scitt/check_operation_status.py (hunk 1 above)
Comment (honourfish) [reply scitt/check_operation_status.py:118]: have handled the timeout exception within main(), so now we don't print the full stack trace, just the correct timeout message and exit with exit code 1.

Also reduced the timeout duration to 60 seconds from 360 seconds (6 minutes) as a merklelog world is much faster :)

File: scitt/check_operation_status.py (hunk 2 above)
Comment (honourfish) [reply scitt/check_operation_status.py]: good suggestion :) done!

**Review by robinbryce — DISMISSED** [review]
LGTM

**Pull request merged**